src/plugins/analysis/strings/code/strings.py

```diff
@@ -11,23 +10,59 @@ class AnalysisPlugin(AnalysisBasePlugin):
     '''
     NAME = 'printable_strings'
     DEPENDENCIES = []
-    DESCRIPTION = 'extracts strings from the files consisting of printable characters'
-    VERSION = '0.2'
+    DESCRIPTION = 'extracts strings and their offsets from the files consisting of printable characters'
+    VERSION = '0.3.3'
+
+    STRING_REGEXES = [
+        (b'[\x09-\x0d\x20-\x7e]{$len,}', 'utf-8'),
+        (b'(?:[\x09-\x0d\x20-\x7e]\x00){$len,}', 'utf-16'),
+    ]
+    FALLBACK_MIN_LENGTH = '8'
 
     def __init__(self, plugin_administrator, config=None, recursive=True, plugin_path=__file__):
         '''
         recursive flag: If True recursively analyze included files
         default flags should be edited above. Otherwise the scheduler cannot overwrite them.
         '''
         self.config = config
+        self.regexes = self._compile_regexes()
+        super().__init__(plugin_administrator, config=config, recursive=recursive, plugin_path=plugin_path)
+
+    def _compile_regexes(self) -> List[Tuple[Pattern[bytes], str]]:
+        min_length = self._get_min_length_from_config()
+        return [
+            (compile(regex.replace(b'$len', min_length.encode())), encoding)
+            for regex, encoding in self.STRING_REGEXES
+        ]
 
-        super().__init__(plugin_administrator, config=config, recursive=recursive, plugin_path=__file__)
+    def _get_min_length_from_config(self):
```

Comment:
is there a reason why we do not use the get function with Fallback Value:
https://docs.python.org/3/library/configparser.html